consumer/src/main/kotlin/streams/kafka/KafkaEventSink.kt

```diff
@@ -113,15 +148,172 @@ class KafkaEventConsumer(private val consumer: KafkaConsumer<String, ByteArray>,
+        if (!records.isEmpty) {
+            try {
+                this.topics.forEach { topic ->
+                    action(topic, records.records(topic).map { JSONUtils.readValue<Any>(it.value()) })
+                }
+            } catch (e: Exception) {
+                // TODO add dead letter queue
+            }
+        }
+    }
+
+    private fun readFromPartition(config: KafkaTopicConfig, action: (String, List<Any>) -> Unit) {
+        setSeek(config.topicPartitionsMap)
+        val records = consumer.poll(0)
+        val consumerRecordsMap = toConsumerRecordsMap(config.topicPartitionsMap, records)
+        if (consumerRecordsMap.isNotEmpty()) {
+            try {
+                consumerRecordsMap.forEach { action(it.key.topic(), it.value.map { JSONUtils.readValue<Any>(it.value()) }) }
+            } catch (e: Exception) {
+                // TODO add dead letter queue
+            }
+        }
+    }
+
+    override fun read(topicConfig: Map<String, Any>, action: (String, List<Any>) -> Unit) {
+        val kafkaTopicConfig = KafkaTopicConfig.fromMap(topicConfig)
+        if (kafkaTopicConfig.topicPartitionsMap.isEmpty()) {
+            readSimple(action)
+        } else {
+            readFromPartition(kafkaTopicConfig, action)
+        }
+    }
+
+    fun toConsumerRecordsMap(topicPartitionsMap: Map<TopicPartition, Long>,
+                             records: ConsumerRecords<String, ByteArray>)
+            : Map<TopicPartition, List<ConsumerRecord<String, ByteArray>>> = topicPartitionsMap
+                .mapValues {
+                    records.records(it.key)
+                }
+                .filterValues { it.isNotEmpty() }
+
+    fun setSeek(topicPartitionsMap: Map<TopicPartition, Long>) {
+        if (isSeekSet) {
+            return
+        }
+        isSeekSet = true
+        consumer.poll(0) // dummy call see: https://stackoverflow.com/questions/41008610/kafkaconsumer-0-10-java-api-error-message-no-current-assignment-for-partition
+        topicPartitionsMap.forEach {
+            when (it.value) {
+                -1L -> consumer.seekToBeginning(listOf(it.key))
+                -2L -> consumer.seekToEnd(listOf(it.key))
+                else -> consumer.seek(it.key, it.value)
+            }
+        }
+    }
+}
+
+class KafkaManualCommitEventConsumer(private val config: KafkaSinkConfiguration,
+                                     private val log: Log): KafkaAutoCommitEventConsumer(config, log) {
+
+    private val topicPartitionOffsetMap = ConcurrentHashMap<TopicPartition, OffsetAndMetadata>()
+
+    override fun start() {
+        if (topics.isEmpty()) {
+            log.info("No topics specified Kafka Consumer will not started")
+            return
+        }
+        this.consumer.subscribe(topics, StreamsConsumerRebalanceListener(topicPartitionOffsetMap, consumer, config.autoOffsetReset, log))
+    }
+
+    private fun readSimple(action: (String, List<Any>) -> Unit): Map<TopicPartition, OffsetAndMetadata> {
```

Comment:
these two methods have almost the same code, is there any way that we can unify them?